src/libraries/Common/src/System/Security/Cryptography/MLKem.cs

```diff
@@ -83,11 +78,7 @@ protected MLKem(MLKemAlgorithm algorithm)
         /// </exception>
         public static MLKem GenerateKey(MLKemAlgorithm algorithm)
         {
-            if (algorithm is null)
-            {
-                throw new ArgumentNullException(nameof(algorithm));
-            }
-
+            ThrowIfNull(algorithm);
             ThrowIfNotSupported();
```

Comment:
I don't know that I particularly care which order we do these, but when we're "done" it should be the same order for all of the things we add in this wave.

Reply:
Is that open-thinking, or are you pointing out that it inconsistent in `MLKem`, or inconsistent between `MLKem` and `{ML,Slh}Dsa`?

Reply:
It's different in your PR and @PranavSenthilnathan 's

https://github.com/dotnet/runtime/pull/113766/files#diff-9086044d530e3b5adf91b93607cb4af6236cad5bbbd20217245aa48a49c3159eR700-R708

So until one of the two is merged, neither is "against precedent", but there are conflicting precedent proposals in flight :smile:.

Clearly I should throw in option 3: we throw an AggregateException that contains both of them! &lt;/sarcasm&gt;

Reply:
> So until one of the two is merged

Well, mine was already merged, this is just changing the throw syntax.

Reply:
See? 😄 

https://github.com/dotnet/runtime/blob/44c08b874792487f81c546ea112b7d4a677c911e/src/libraries/Common/src/System/Security/Cryptography/MLKem.cs#L86-L91

However, mine was very much intentional. SHA-3 does it in that order, too:

https://github.com/dotnet/runtime/blob/44c08b874792487f81c546ea112b7d4a677c911e/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SHA3_256.cs#L174-L183

Reply:
My basic preflight order is:

1. ArgumentExceptions (null first, then any logical one)
2. PNSE
3. ObjectDisposedException
4. "Do the thing" exceptions (CryptographicException, InvalidOperationExceptions, etc)

Reply:
Fair enough.  Left a comment on the other PR.